lib/pipelines/code-pipeline.ts

```diff
@@ -48,10 +69,15 @@ export type PipelineProps = {
     owner: string;
 
     /**
-     * Repository for the pipeline
+     * Repository for the pipeline (GitHub handle).
      */
     repository: GitHubSourceRepository;
 
+    /**
+     * CodeCommit repository properties for the pipeline (CodeCommit handle).
+     */
+    codeCommitProps: CodeCommitProps;
```

Comment:
Looks like both codeCommitProps and repository fields are required while they are mutually exclusive.